apps/src/applab/applab.js

```diff
@@ -1398,6 +1404,68 @@ Applab.showConfirmationDialog = function(config) {
   dialog.show();
 };
 
+/**
+ * Get a testResult and message value based on an examination of the
+ * executionLog against the level's logConditions
+ *
+ * @returns {!Object}
+ */
+Applab.getResultsFromLog = function () {
```

Comment:
What if this took an execution log as an input, i.e.
```
Applab.getResultsFromLog = function (executionLog) {
...
```
Then elsewhere we would call: `Applab.getResultsFromLog(Applab.currentExecutionLog)`.

Doing so would make this much easier to test, as I believe we would have a pure function at that point.

Incidentally, this seems like it would be a great function to write some tests for :)